components/script/dom/htmlimageelement.rs

```diff
@@ -356,20 +390,244 @@ impl HTMLImageElement {
     }
 
     /// <https://html.spec.whatwg.org/multipage/#update-the-source-set>
-    fn update_source_set(&self) -> Vec<DOMString> {
+    fn update_source_set(&self) {
+        // Step 1
+        *self.source_set.borrow_mut() = HTMLImageElement::new_source_set();
+
+        // Step 2
         let elem = self.upcast::<Element>();
-        // TODO: follow the algorithm
-        let src = elem.get_string_attribute(&local_name!("src"));
-        if src.is_empty() {
-            return vec![]
+        let parent = elem.upcast::<Node>().GetParentElement();
+        let is_parent_picture = match parent.clone() {
+            Some(xparent) => xparent.is::<HTMLPictureElement>(),
+            None => false
+        };
+        let nodes = parent.unwrap().upcast::<Node>().ChildNodes();
```

Comment:
This looks like it would break if used on any image element that does not have a parent. Let's do this instead:
```rust
let elements = match parent {
    Some(p) => {
        let nodes = p.upcast::<Node>().ChildNodes();
        nodes.iter().filter_map(DomRoot::downcast::<Element>)
            .map(|n| DomRoot::from_ref(&*n)).collect()
    }
    None => {
        vec![DomRoot::from_ref(&*elem)]
    }
};
````

Reply:
Keeping nodes inside the Some{ } gives "borrowed value does not live long enough" error. That is the reason it was kept outside before. 😅 Any other approach to solve this error ?

Reply:
What specific part of the code is the error message referring to?